nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,524 @@
+        The sympa.conf configuration file as key value set.
+      '';
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    services.sympa.settings = (mapAttrs (_: v: mkDefault v) {
+      domain     = if cfg.mainDomain != null then cfg.mainDomain else (head fqdns);
+      listmaster = concatStringsSep "," cfg.listMasters;
+      lang       = cfg.lang;
+
+      home       = "${dataDir}/list_data";
+      arc_path   = "${dataDir}/arc";
+      bounce_path = "${dataDir}/bounce";
+
+      sendmail = "/run/wrappers/bin/sendmail";
+
+      db_type = cfg.database.type;
+      db_name = cfg.database.name;
+      db_passwd = "#dbpass#";
+    }
+    // (optionalAttrs (cfg.database.host != null) {
+      db_host = cfg.database.host;
+    })
+    // (optionalAttrs (cfg.database.port != null) {
+      db_port = cfg.database.port;
+    })
+    // (optionalAttrs (cfg.database.user != null) {
+      db_user = cfg.database.user;
+    })
+    // (optionalAttrs (cfg.mta.type == "postfix") {
+      sendmail_aliases = "${dataDir}/sympa_transport";
+      aliases_program  = "${pkgs.postfix}/bin/postmap";
+      aliases_db_type  = "hash";
+    })
+    // (optionalAttrs cfg.web.enable {
+      static_content_path = "${dataDir}/static_content";
+      css_path            = "${dataDir}/static_content/css";
+      pictures_path       = "${dataDir}/static_content/pictures";
+      mhonarc             = "${pkgs.perlPackages.MHonArc}/bin/mhonarc";
+    }));
+
+    environment = {
+      systemPackages = [ pkg ];
+    };
+
+    users.users = optional (cfg.user == "sympa")
+      { name = cfg.user;
+        description = "Sympa mailing list manager user";
+        group =  cfg.group;
+        home = cfg.dataDir;
+        createHome = true;
+        isSystemUser = true;
+      };
+
+    users.groups =
+      optional (cfg.group == "sympa")
+      { name = cfg.group;
+      };
+
+    warnings = optional (cfg.database.password != "")
+      ''config.services.sympa.database.password will be stored as plaintext
+        in the Nix store. Use database.passwordFile instead.'';
+
+    # Create database passwordFile default when password is configured.
+    services.sympa.database.passwordFile =
+      (mkDefault (toString (pkgs.writeTextFile {
+        name = "sympa-database-password";
+        text = cfg.database.password;
+      })));
+
+    systemd.services.sympa = {
+      description = "Sympa mailing list manager";
+
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      wants = sympaSubServices;
+      before = sympaSubServices;
+      restartTriggers = [ mainConfig cfg.database.passwordFile ];
+      serviceConfig = sympaServiceConfig "sympa_msg";
+
+      preStart = ''
+        umask 0077
+        mkdir -p ${dataDir}/etc
+        mkdir -p ${dataDir}/spool
+        mkdir -p ${dataDir}/list_data
+        mkdir -p ${dataDir}/arc
+        mkdir -p ${dataDir}/bounce
+
+        cp -f ${mainConfig} ${dataDir}/etc/sympa.conf
```

Comment:
You should create these directories and link this file with the more declarative `systemd.tmpfiles` instead (which contrary to its name is used for more than just temporary files). Same for other `mkdir`'s, `touch`'s and `chmod`'s in this script.

Reply:
Converted some of the operations to tmpfiles rules.

Reply:
Converted more file copies to tmpfiles rules (everything using `settingsFile`).